File: launch/drm.indirect/src/main/java/org/westmalle/compositor/bootstrap/drm/indirect/IndirectDrmEglCompositor.java
package org.westmalle.compositor.bootstrap.drm.indirect;


import dagger.Component;
import org.westmalle.compositor.core.CoreModule;
import org.westmalle.compositor.core.KeyBindingFactory;
import org.westmalle.compositor.core.LifeCycle;
import org.westmalle.compositor.drm.egl.DrmEglPlatformModule;
import org.westmalle.compositor.gles2.Gles2RendererModule;
import org.westmalle.compositor.input.LibinputSeatFactory;
import org.westmalle.launch.indirect.IndirectModule;
import org.westmalle.tty.Tty;
import org.westmalle.tty.TtyModule;

import javax.inject.Singleton;

@Singleton
@Component(modules = {CoreModule.class,
                      Gles2RendererModule.class,
                      DrmEglPlatformModule.class,
                      TtyModule.class,
                      IndirectModule.class})
public interface IndirectDrmEglCompositor {
    LifeCycle lifeCycle();

    LibinputSeatFactory seatFactory();

    KeyBindingFactory keyBindingFactory();

    Tty tty();
}

File: launch.drm.indirect/src/main/kotlin/org/westford/compositor/launch/drm/indirect/IndirectDrmEglCompositor.kt
package org.westford.compositor.launch.drm.indirect

import dagger.Component
import org.westford.compositor.core.CoreModule
import org.westford.compositor.core.KeyBindingFactory
import org.westford.compositor.core.LifeCycle
import org.westford.compositor.drm.egl.DrmEglPlatformModule
import org.westford.compositor.gles2.Gles2RendererModule
import org.westford.compositor.input.LibinputSeatFactory
import org.westford.launch.indirect.IndirectModule
import org.westford.tty.Tty
import org.westford.tty.TtyModule

import javax.inject.Singleton

@Singleton @Component(modules = arrayOf(CoreModule::class,
                                        Gles2RendererModule::class,
                                        DrmEglPlatformModule::class,
                                        TtyModule::class,
                                        IndirectModule::class)) interface IndirectDrmEglCompositor {
    fun lifeCycle(): LifeCycle

    fun seatFactory(): LibinputSeatFactory

    fun keyBindingFactory(): KeyBindingFactory

    fun tty(): Tty
}
